contracts/Aavegotchi/facets/ERC721BuyOrderFacet.sol

```diff
@@ -0,0 +1,190 @@
+            require(erc721BuyOrder.timeCreated != 0, "ERC721BuyOrder: ERC721 buyOrder does not exist");
+            require((erc721BuyOrder.cancelled == false) && (erc721BuyOrder.timePurchased == 0), "ERC721BuyOrder: Already processed");
+
+            LibBuyOrder.cancelERC721BuyOrder(oldBuyOrderId);
+        }
+
+        // Transfer GHST
+        LibERC20.transferFrom(s.ghstContract, sender, address(this), _priceInWei);
+
+        // Place new buy order
+        s.nextERC721BuyOrderId++;
+        uint256 buyOrderId = s.nextERC721BuyOrderId;
+
+        s.erc721TokenToBuyOrderIdIndexes[_erc721TokenAddress][_erc721TokenId][buyOrderId] = s
+        .erc721TokenToBuyOrderIds[_erc721TokenAddress][_erc721TokenId].length;
+        s.erc721TokenToBuyOrderIds[_erc721TokenAddress][_erc721TokenId].push(buyOrderId);
+        s.buyerToBuyOrderId[_erc721TokenAddress][_erc721TokenId][sender] = buyOrderId;
+
+        s.erc721BuyOrders[buyOrderId] = ERC721BuyOrder({
+            buyOrderId: buyOrderId,
+            buyer: sender,
+            erc721TokenAddress: _erc721TokenAddress,
+            erc721TokenId: _erc721TokenId,
+            priceInWei: _priceInWei,
+            validationHash: LibBuyOrder.generateValidationHash(_erc721TokenAddress, _erc721TokenId, _validationOptions),
+            timeCreated: block.timestamp,
+            timePurchased: 0,
+            cancelled: false,
+            validationOptions: _validationOptions
+        });
+        emit ERC721BuyOrderAdd(buyOrderId, sender, _erc721TokenAddress, _erc721TokenId, category, _priceInWei, block.timestamp);
+    }
+
+    function cancelERC721BuyOrder(uint256 _buyOrderId) external {
+        address sender = LibMeta.msgSender();
+        ERC721BuyOrder memory erc721BuyOrder = s.erc721BuyOrders[_buyOrderId];
+
+        require(erc721BuyOrder.timeCreated != 0, "ERC721BuyOrder: ERC721 buyOrder does not exist");
+        require(
+            (sender == s.aavegotchis[erc721BuyOrder.erc721TokenId].owner) || (sender == erc721BuyOrder.buyer),
+            "ERC721BuyOrder: Only aavegotchi owner or buyer can call this function"
+        );
+        require((erc721BuyOrder.cancelled == false) && (erc721BuyOrder.timePurchased == 0), "ERC721BuyOrder: Already processed");
+
+        LibBuyOrder.cancelERC721BuyOrder(_buyOrderId);
+    }
+
+    function executeERC721BuyOrder(uint256 _buyOrderId) external {
+        address sender = LibMeta.msgSender();
+        ERC721BuyOrder memory erc721BuyOrder = s.erc721BuyOrders[_buyOrderId];
+
+        require(erc721BuyOrder.timeCreated != 0, "ERC721BuyOrder: ERC721 buyOrder does not exist");
+        require(sender == s.aavegotchis[erc721BuyOrder.erc721TokenId].owner, "ERC721BuyOrder: Only aavegotchi owner can call this function");
+        require((erc721BuyOrder.cancelled == false) && (erc721BuyOrder.timePurchased == 0), "ERC721BuyOrder: Already processed");
+
+        // disable for gotchi in lending
```

Comment:
Is this code necessary? Gotchi Lending should already prevent the gotchi transfer from happening right?

Reply:
Yes, it needed.
When gotchi is in lending, owner is lender, so we need to check it.

Reply:
So it's not possible to create an offer on a gotchi currently being borrowed, because the owner is the lender? 

What if we used the original owner instead of the current owner (borrower)?

Reply:
The owner of the gotchi wouldn't be able to accept the offer until the Gotchi is returned, but at least people could make offers on it.

Reply:
This is execute function. this should be accepted after Gotchi is returned.
In ordering function, there's no check lending status. So buy order could be placed for lending gotchis.

Reply:
Got it. Maybe we could replace that code with this getter function then? 

```  function isAavegotchiLent(uint32 _erc721TokenId) external view returns (bool) {
        return LibGotchiLending.isAavegotchiLent(_erc721TokenId);
    }```

Reply:
OK, let me update.

Reply:
Done.